sdk/search/azure-search/src/main/java/com/azure/search/documents/util/RegexFlagsUtils

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.util;
+
+import com.azure.search.documents.models.RegexFlags;
+import java.util.Arrays;
+
+/**
+ * Utility class for {@link RegexFlags} complex operations.
+ */
+public final class RegexFlagsUtils {
+
+    /**
+     * Combines a list of {@link RegexFlags} into one separated by "|".
+     * E.g. {@code "CANON_EQ|CASE_INSENSITIVE|COMMENTS"}
+     *
+     * @param regexFlags A list a {@link RegexFlags} to combine.
+     * @return New combined {@link RegexFlags} separated by vertical bar.
+     */
+    public static RegexFlags combineRegexFlags(RegexFlags... regexFlags) {
```

Comment:
I'm a bit worried that `RegexFlags` is an `ExpandableStringEnum` as this may grow unbounded with the number of permutations available and will remain in memory as it is backed by a static dictionary. 

@brjohnstmsft does the ordering of the regex flags matter? If not we should sort before joining as that reduces the number of possible combinations that will be stored in static memory.

Reply:
The ordering shouldn't matter, but maybe we should consider modeling the `flags` properties of `PatternAnalyzer` and `PatternTokenizer` as collections instead of introducing this combining primitive? I believe @xirzec is taking this approach in TypeScript.

Reply:
I have considered to do this. These two classes are final auto-gen classes, we have to do some transformation work if adding this function over there.

Reply:
Sync with @xirzec , I will move this function to the best place after discussing with Jeff.